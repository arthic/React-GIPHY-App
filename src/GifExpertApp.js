import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = ['xbox series x'] }) => {
	// const categories = ['Naruto', 'Goku', 'Xbox']
	/* const [categories, setCategories] = useState([
			'DMT'
		]) */
	const [categories, setCategories] = useState(
		defaultCategories
	)
	return(
		<>
			<section id="search">
				<h2 className="animate__animated animate__bounceInDown">GIPHY App</h2>
				<div className="powered">
					<p>Powered by</p>
					<a href="https://giphy.com/" target="_blank" rel="noopener noreferrer">
						<img alt="giphy" src="./assets/giphy.png" />
					</a>
				</div>
				<AddCategory setCategories={setCategories}/>
			</section>
			{/* <button onClick={handleAdd}>Agregar</button> */}
			<div className="border-gradient">
			</div>
			<ol>
			{
				categories.map(category =>
					<GifGrid
					className="animate__animated animate__bounceInLeft"
					key={category}
					category={category}
					/>)
			}
			</ol>
		</>
	)
}
export default GifExpertApp
