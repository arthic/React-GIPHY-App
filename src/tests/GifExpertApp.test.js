import React from 'react'
import '@testing-library/jest-dom'
import {shallow} from 'enzyme'
import GifExpertApp from '../GifExpertApp'

describe('Testing de <GifExpertApp />', () => {

	test('Despliegue correcto de <GifExpertApp />', () => {
		const wrapper = shallow(<GifExpertApp />)
		expect(wrapper).toMatchSnapshot()
	})
	test('Debe mostrar la lista de categorias', () => {
		const categories = ['Test-01', 'Test-02']
		const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)
		expect(wrapper.find('GifGrid').length).toBe(categories.length)
		expect(wrapper).toMatchSnapshot()
	})
});

