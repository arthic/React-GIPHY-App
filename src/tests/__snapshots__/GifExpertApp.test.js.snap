// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing de <GifExpertApp /> Debe mostrar la lista de categorias 1`] = `
<Fragment>
  <section
    id="search"
  >
    <h2
      className="animate__animated animate__bounceInDown"
    >
      GIPHY App
    </h2>
    <div
      className="powered"
    >
      <p>
        Powered by
      </p>
      <a
        href="https://giphy.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="giphy"
          src="./assets/giphy.png"
        />
      </a>
    </div>
    <AddCategory
      setCategories={[Function]}
    />
  </section>
  <div
    className="border-gradient"
  />
  <ol>
    <GifGrid
      category="Test-01"
      className="animate__animated animate__bounceInLeft"
      key="Test-01"
    />
    <GifGrid
      category="Test-02"
      className="animate__animated animate__bounceInLeft"
      key="Test-02"
    />
  </ol>
</Fragment>
`;

exports[`Testing de <GifExpertApp /> Despliegue correcto de <GifExpertApp /> 1`] = `
<Fragment>
  <section
    id="search"
  >
    <h2
      className="animate__animated animate__bounceInDown"
    >
      GIPHY App
    </h2>
    <div
      className="powered"
    >
      <p>
        Powered by
      </p>
      <a
        href="https://giphy.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="giphy"
          src="./assets/giphy.png"
        />
      </a>
    </div>
    <AddCategory
      setCategories={[Function]}
    />
  </section>
  <div
    className="border-gradient"
  />
  <ol>
    <GifGrid
      category="xbox series x"
      className="animate__animated animate__bounceInLeft"
      key="xbox series x"
    />
  </ol>
</Fragment>
`;
